content/languages/javascript/authoring.md

```diff
@@ -0,0 +1,174 @@
+
+## Description
+
+JavaScript code blocks can be inserted with JavaScript-specific part in [sequential code blocks](/references/markdown/extensions/#sequential-code-blocks):
+
+~~~
+```javascript
+
+...your code here...
+
+```
+~~~
+
+JavaScript-specific paragraphs can be inserted with [language conditional rendering](/references/markdown/extensions/#conditional-rendering):
+
+```
+~~~if:javascript
+
+...text visible only for JavaScript description...
+
+~~~
+
+~~~if-not:javascript
+
+...text not visible in JavaScript description...
+
+~~~
+```
+
+## Tasks and Requirements
+
+Some concepts don't always translate well to or from JavaScript. Because of this, some constructs should be avoided and some translations just shouldn't be done.
+- Avoid returning different data types depending on the situation (_"Return the result, or the string 'Error' if no result can be found..."_). JavaScript is dynamically typed, which is not the case for some other languages. Returning `null` might be appropriate in some situations, but throwing an error might be better in others.
+
+_TODO: Finish this_
+
+## Tests
+
+### Testing framework
+
+
+JavaScript kata should use [Mocha](https://mochajs.org/) to implement and execute tests, and [Chai](https://www.chaijs.com/) as the assertion library.
+
+:::note
+For simplicity, through the rest of this article it's assumed that the `chai` package is imported and `assert` is extracted from it with the following statement:
+
+```javascript
+const chai = require('chai');
+const assert = chai.assert;
+```
+
+This is a convention used in many JavaScript kata, but it's not a requirement, and authors can choose to import the package in any way they find suitable for them. Also, the `should` or `expect` style can also be used from `chai`.
+:::
+
+The most commonly used assertions include:
+
+- [`assert.strictEqual(actual, expected[, message])`](https://www.chaijs.com/api/assert/#method_strictequal) - Asserts that `actual === expected`.
+- [`assert.deepEqual(actual, expected[, message])`](https://www.chaijs.com/api/assert/#method_deepequal) - Asserts that two arrays or objects have the same values. `strictEqual` does not work since arrays and objects are reference types in JS.
+- [`assert.closeTo(actual, expected, delta[, message])`](https://www.chaijs.com/api/assert/#method_closeto) - Asserts that `abs(actual - expected) <= delta`. Should be used when comparing floating-point numbers.
+
+See the [Chai docs](https://www.chaijs.com/api/assert/) for all the methods.
+
+:::note
+When comparing arrays over 40 elements long, [the `truncateThreshold` option](https://www.chaijs.com/guide/styles/#configtruncatethreshold) should be set to `0` to make sure that the arrays will not be truncated in an error message. If the array is hundreds/thousands of elements long, do not do this as it will flood the console.
```

Comment:
40 seems to be somewhat specific here, and probably a configuration detail. Maybe just mention that `truncateThreshold` option exists, explain what it controls, and why authors should be careful?

```suggestion
The [`truncateThreshold` option](https://www.chaijs.com/guide/styles/#configtruncatethreshold) controls how arrays are stringified when `expected` and `actual` values are prepared for display in assertion messages. Setting it to 0 provides the most detailed feedback even for large arrays, but then assertion messages can flood the console and make the output difficult to read.
```